### PR TITLE
fix: striped table when element selected is text/dropdown/multiselect/editable/datepicker

### LINKED ISSUE #741: [Widget]: On selection of tags option in table category type, page crashes
Steps to reproduce: 

1. Drag and drop a table
2. Open widget inspector
3. Select the tags option from the dropdown in column type
4. Page breaks

[tags_error.mov.zip](https://github.com/ToolJet/ToolJet/files/7154333/tags_error.mov.zip)

### PR DESCRIPTION
**Pull request by gondar00**
This PR fixes the table design when combination of a field text + table style striped

Before: 
image

After:
image

-----

Before: 
image
After: 
image

----

Fixes #741 

Incorrect value was provided in tags selection

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
image

Steps to reproduce: 
a) drag and drop a table widget 
b) set the `email` column as editable 

Can you please take a look? @gondar00

**Comment by gondar00** [comment]
I think the above image would've been on a diff. branch

On this branch it looks like the below: 
image

**Comment by gondar00** [comment]
1. Making a column editable
2. Making a column type as dropdown 

These I am working on currently

**Comment by gondar00** [comment]
Making a column editable is fixed, 

pending is making a column type as datepicker

**Comment by gondar00** [comment]
Datepicker fix is done

**Comment by sherfin94** [comment]
Looks like we didn't cover action buttons

https://www.loom.com/share/7340770dfe304a9db8c8cbbdac98effc

**Comment by gondar00** [comment]
@sherfin94 yeah added fix for action button 👍

**Comment by sherfin94** [comment]
There seems to be a strange issue with drop-down, multiselect, badge etc on this branch:
https://www.loom.com/share/3f660dddd2114bb3a508a23a18432e10

A similar issue seems to be there with datepicker also, though its certainly better than what we already have with datepicker :grimacing: .
image

**Comment by gondar00** [comment]
image

```

  td {
    min-height: 35px;
    overflow-x: initial; // so due to overflow being initial the content overflows from the box in case of many fields
    margin: auto;
    ....
}
```

**Comment by gondar00** [comment]
image
```

  td {
    min-height: 35px;
    overflow-x: auto; // i added auto here so that a scrollbar comes, but that has broken select/dropdown 
    margin: auto;
    ....
}
```

**Comment by gondar00** [comment]
I'll change it back to `initial` for now

File: frontend/src/Editor/Components/Table/Table.jsx [hunk 1]
```diff
@@ -296,19 +297,96 @@ export function Table({
...
+                          column.key || column.name,
+                          e.target.value,
+                          cell.row.original
+                        );
+                      }
+                    }}
+                    className={`form-control-plaintext form-control-plaintext-sm ${!isValid ? 'is-invalid' : ''}`}
+                    defaultValue={cellValue}
+                  />
+                  <div class="invalid-feedback">{validationError}</div>
```
Comment (sherfin94) [inline frontend/src/Editor/Components/Table/Table.jsx:369]:
```suggestion
                  <div className="invalid-feedback">{validationError}</div>
```

**Comment by sherfin94** [comment]
Went through the code, looks nice!.
There is this one last problem I missed: It also has to support proper stripes when validation text is present
image

Once that's fixed, could you also merge the latest `develop` to this?

**Review by sherfin94 — CHANGES_REQUESTED** [review]

**Review by sherfin94 — APPROVED** [review]

**Pull request merged**